src/openeo_gfmap/features/feature_extractor.py

```diff
@@ -0,0 +1,291 @@
+"""Feature extractor functionalities. Such as a base class to assist the
+implementation of feature extractors of a UDF.
+
+Be careful: a lot of
+  __    __  ______  ______  ______  ______  ______  ______  ______  ______  ______  __    __  __    __  __  __   __  ______
+ /\ "-./  \/\  ___\/\__  _\/\  __ \/\  == \/\  == \/\  __ \/\  ___\/\  == \/\  __ \/\ "-./  \/\ "-./  \/\ \/\ "-.\ \/\  ___\
+ \ \ \-./\ \ \  __\\/_/\ \/\ \  __ \ \  _-/\ \  __<\ \ \/\ \ \ \__ \ \  __<\ \  __ \ \ \-./\ \ \ \-./\ \ \ \ \ \-.  \ \ \__ \
+  \ \_\ \ \_\ \_____\ \ \_\ \ \_\ \_\ \_\   \ \_\ \_\ \_____\ \_____\ \_\ \_\ \_\ \_\ \_\ \ \_\ \_\ \ \_\ \_\ \_\\"\_\ \_____\
+   \/_/  \/_/\/_____/  \/_/  \/_/\/_/\/_/    \/_/ /_/\/_____/\/_____/\/_/ /_/\/_/\/_/\/_/  \/_/\/_/  \/_/\/_/\/_/ \/_/\/_____/
+
+in here
+"""
+
+REQUIRED_IMPORTS = """
+import inspect
+from abc import ABC, abstractmethod
+
+import openeo
+from openeo.udf import XarrayDataCube, inspect
+from openeo.udf.run_code import execute_local_udf
+from openeo.udf.udf_data import UdfData
+
+import pyproj
+import xarray as xr
+import numpy as np
+
+from typing import Union
+"""
+
+exec(REQUIRED_IMPORTS)
```

Comment:
I would not do that, this will cause too many problems with linting tools (e.g. like the GH actions that are currently failing because of this I think)

Reply:
Yes I already encountered one, I changed that by importing manually.